lib/logstash/outputs/s3.rb

```diff
@@ -398,12 +398,14 @@ def restore_from_crash
 
     temp_folder_path = Pathname.new(@temporary_directory)
     Dir.glob(::File.join(@temporary_directory, "**/*"))
-      .select { |file| ::File.file?(file) }
-      .each do |file|
-      temp_file = TemporaryFile.create_from_existing_file(file, temp_folder_path)
+      .select { |file_path| ::File.file?(file_path) }
```

Comment:
We will need to figure out a way between the recovery code and this glob to ignore "recovery in progress" - currently, in the event of a failure/crash during "recovery in progress", this `glob` definition will pick up the original file to be recovered, plus any files in mid-recovery - ie `ls.s3<>part0.txt.gz`, `ls.s3<>.part0-recovered.txt.gz`, and `ls.s3<>.part0-recovered.txt.gz`. This can lead to a few different outcomes - duplicate data being uploaded to S3, potentially multiple `-recovered.txt` and `-recovered.txt.gz` extensions, and, in the event one of these files is empty, `clean_temporary_file(temp_file)` will be called, which deletes the entire sub directory that all of these files are stored in.

Reply:
Is `Dir.glob().select{}` async for this case?
The possible cases would be 
- multiple LS instances are accessing to same S3 temp path but as we discussed offline the recommendation would be providing a separate path for each instance.
- unrecoverable zip files which always stay

So, I have added a logic where figures out the under recovery process files (`under_recovery_files`) and skips the file if it is in this list.